core/vm/logger.go

```diff
@@ -200,15 +206,45 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 			)
 			l.storage[contract.Address()][address] = value
 			storage = l.storage[contract.Address()].Copy()
+
+			extraData = types.NewExtraData()
+			if err := traceStorageProof(l, scope, extraData); err != nil {
+				log.Warn("Failed to get proof", "contract address", contract.Address().String(), "key", address.String(), "err", err)
+			}
```

Comment:
what about we 
1) always traceStorageProof for `SLOAD` & `SSTORE`, and only traceStorageProofAfter for `SSTORE`

for now we can doing
2) always only traceStorageProof for `SSTORE`, and traceStorageProofAfter for `SLOAD` & `SSTORE`

To me 1) looks more clean. but not a big problem, doesn't really matter